src/ImageSharp/Formats/Jpeg/Components/Decoder/ComponentProcessors/DownScalingComponentProcessor8.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Runtime.CompilerServices;
+using SixLabors.ImageSharp.Memory;
+
+namespace SixLabors.ImageSharp.Formats.Jpeg.Components.Decoder
+{
+    internal sealed class DownScalingComponentProcessor8 : ComponentProcessor
+    {
+        private readonly float dcDequantizer;
+
+        public DownScalingComponentProcessor8(MemoryAllocator memoryAllocator, JpegFrame frame, IRawJpegData rawJpeg, Size postProcessorBufferSize, IJpegComponent component)
+            : base(memoryAllocator, frame, postProcessorBufferSize, component, 1)
+            => this.dcDequantizer = rawJpeg.QuantizationTables[component.QuantizationTableIndex][0];
+
+        public override void CopyBlocksToColorBuffer(int spectralStep)
+        {
+            Buffer2D<Block8x8> spectralBuffer = this.Component.SpectralBlocks;
+
+            float maximumValue = this.Frame.MaxColorChannelValue;
+            float normalizationValue = MathF.Ceiling(maximumValue / 2);
+
+            int destAreaStride = this.ColorBuffer.Width;
+
+            int blocksRowsPerStep = this.Component.SamplingFactors.Height;
+            Size subSamplingDivisors = this.Component.SubSamplingDivisors;
+
+            int yBlockStart = spectralStep * blocksRowsPerStep;
+
+            for (int y = 0; y < blocksRowsPerStep; y++)
+            {
+                int yBuffer = y * this.BlockAreaSize.Height;
+
+                Span<float> colorBufferRow = this.ColorBuffer.DangerousGetRowSpan(yBuffer);
+                Span<Block8x8> blockRow = spectralBuffer.DangerousGetRowSpan(yBlockStart + y);
+
+                for (int xBlock = 0; xBlock < spectralBuffer.Width; xBlock++)
+                {
+                    // get direct current term - averaged 8x8 pixel value
+                    float dc = blockRow[xBlock][0];
+
+                    // dequantization
+                    dc *= this.dcDequantizer;
+
+                    // Normalize & round
+                    dc = (float)Math.Round(Numerics.Clamp(dc + normalizationValue, 0, maximumValue));
+
+                    // Save to the intermediate buffer
+                    int xColorBufferStart = xBlock * this.BlockAreaSize.Width;
+                    ScaledCopyTo(
+                        dc,
+                        ref colorBufferRow[xColorBufferStart],
+                        destAreaStride,
+                        subSamplingDivisors.Width,
+                        subSamplingDivisors.Height);
+                }
+            }
+        }
+
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static void ScaledCopyTo(float value, ref float destRef, int destStrideWidth, int horizontalScale, int verticalScale)
+        {
+            if (horizontalScale == 1 && verticalScale == 1)
+            {
+                Unsafe.Add(ref destRef, 0) = value;
```

Comment:
`Unsafe.Add` with zero argument is removed by JIT. Explicit zero addition is IMO more self-explanatory in case of multiple lines like this:

```cs
dest = ...;
Unsafe.Add(ref dest, 1) = ...;
Unsafe.Add(ref dest, 2) = ...;
Unsafe.Add(ref dest, 3) = ...;
```

But this is a one-liner so it's redundant here.

Reply:
Yep, but IL-size is also a metric. I don't know how relevant it is, so I have no strong opinion here.
For parity with the lines below, leave it as is?

Reply:
IL-size matters, true, I'll take a look at this once again for refactoring before making this PR ready for merging.